src/cdk-lib/bedrock/knowledge-bases/structured-data-knowledge-base.ts

```diff
@@ -0,0 +1,330 @@
+  USERNAME = 'USERNAME'
+}
+
+/**
+ * Represents a Knowledge Base, either created with CDK or imported.
+ */
+export interface ISqlKnowledgeBase extends IKnowledgeBase {
+
+  /**
+   * Grant the given identity permissions to retrieve content from the knowledge base.
+   */
+  grantRetrieve(grantee: iam.IGrantable): iam.Grant;
+
+  /**
+   * Grant the given identity permissions to retrieve content from the knowledge base and generate.
+   */
+  grantRetrieveAndGenerate(grantee: iam.IGrantable): iam.Grant;
+}
+
+/******************************************************************************
+ *                              ABSTRACT CLASS
+ *****************************************************************************/
+/**
+ * Abstract base class for Sql Knowledge Base.
+ * Contains methods valid for KBs either created with CDK or imported.
+ */
+abstract class SqlKnowledgeBaseBase extends KnowledgeBaseBase implements ISqlKnowledgeBase {
+  public abstract readonly knowledgeBaseArn: string;
+  public abstract readonly knowledgeBaseId: string;
+  public abstract readonly role: iam.IRole;
+  public abstract readonly description?: string;
+  public readonly instruction?: string;
+  public readonly type: KnowledgeBaseType = KnowledgeBaseType.SQL;
+
+  constructor(scope: Construct, id: string) {
+    super(scope, id);
+  }
+}
+
+/******************************************************************************
+ *                        PROPS FOR NEW CONSTRUCT
+ *****************************************************************************/
+/**
+ * Properties for a Sql knowledge base.
+ */
+export interface SqlKnowledgeBaseProps extends CommonKnowledgeBaseProps {
+
+  /**
+   * @default - Amazon Redshift is the only and default query engine for Sql KB for now.
+   */
+  readonly sqlDatabaseType?: SqlDatabaseType;
+
+  /**
+   * The type of query engine, which can be of `SERVERLESS` or `PROVISIONED` types.
+   */
+  readonly redshiftQueryEngineType: RedshiftQueryEngineType;
+
+  /**
+     * The Amazon Redshift provisioned configuration.
+     * If redshiftQueryEngineType is not of type `PROVISIONED`,
+     * do not include this property as it will throw error.
+     */
+  readonly redshiftProvisionedConfiguration?: bedrock.CfnKnowledgeBase.RedshiftProvisionedConfigurationProperty;
+
+  /**
+   * The Amazon Redshift serverless configuration.
+   * If redshiftQueryEngineType is not of type `SERVERLESS`,
+   * do not include this property as it will throw error.
+   */
+  readonly redshiftServerlessConfiguration?: bedrock.CfnKnowledgeBase.RedshiftServerlessConfigurationProperty;
+
+  /**
+   * The storage configuration for AWS Glue Data Catalog.
+   * If redshiftQueryEngineStorageType is not of type `AWS_DATA_CATALOG`,
+   * do not include this property as it will throw error.
+   */
+  readonly redshiftQueryEngineAwsDataCatalogStorageConfiguration?: bedrock.CfnKnowledgeBase
+    .RedshiftQueryEngineAwsDataCatalogStorageConfigurationProperty;
```

Comment:
Same as before, maybe we can create an L2 such as `QueryEngineStorage.fromAwsDataCatalog(tableNames)` and `QueryEngineStorage.fromRedshift(databaseName)` using static methods. Some inspiration can be taken from [here](https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/aws-cloudfront-origins/lib/s3-bucket-origin.ts).